core/ulog/log.go

```diff
@@ -60,21 +60,45 @@ type baselogger struct {
 type Log interface {
 	// Configure Log object with the provided log.Configuration
 	Configure(Configuration)
```

Comment:
I would argue that Configuration, SetLogger, and SetLevel methods should be in some logger factory or builder, not in the Log interface itself. Constructing a logger and actually logging are two different and independent concerns, they do not need to be in the same interface. It widens the API surface and opens it for abuse, e.g. you can pass a Debug logger to some library, but the library can override the log level, which it should not be able to do.